tests/callout.c

```diff
@@ -57,35 +53,39 @@ static int test_callout_simple(void) {
 
 /* This test checks if the order of execution for scheduled callouts is correct.
  */
-static int current = 0;
+
+#define ORDER_N 10
+static int order[ORDER_N] = {2, 5, 4, 6, 9, 0, 8, 1, 3, 7};
+static bool processed[ORDER_N];
 
 static void callout_ordered(void *arg) {
+  /* There is no race condition here as callouts run in bottom half (with
+   * interrupts disabled). */
   assert(intr_disabled());
-  /* This incrementation is safe as callouts run with interrupts disabled. */
-  current++;
 
-  if (current == 10)
-    sleepq_signal(callout_ordered);
+  int ord = (int)arg;
+  if (ord > 0)
+    assert(processed[ord - 1]);
```

Comment:
Actually I think we could make a better assert: keep the `current` variable and `assert(ord == curr)` (+-1 depending on  the order of instructions).

Reply:
Done.